src/loops.rs

```diff
@@ -69,6 +72,37 @@ impl LintPass for LoopsPass {
                 }
             }
 
+            // if this for-loop is iterating over a two-sided range...
+            if let ExprRange(Some(ref start_expr), Some(ref stop_expr)) = arg.node {
+                // and both sides are literals...
+                if let ExprLit(ref start_lit) = start_expr.node {
+                    if let ExprLit(ref stop_lit) = stop_expr.node {
+                        // and they are both integers...
+                        if let LitInt(start_idx, _) = start_lit.node {
+                            if let LitInt(stop_idx, _) = stop_lit.node {
+                                // and the start index is greater than the stop index,
+                                // this loop will never run. This is often confusing for developers
+                                // who think that this will iterate from the larger value to the
+                                // smaller value.
+                                if start_idx > stop_idx {
+                                    span_lint(cx, REVERSE_RANGE_LOOP, expr.span, &format!(
+                                        "this range is empty and this for loop will never run. \
```

Comment:
Might be better to use [`span_help_and_lint`](https://github.com/Manishearth/rust-clippy/blob/master/src/utils.rs#L203) and put the suggestion in the help part.